File: flutter/lib/screens/settings_screen.dart
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';

class SettingsScreen extends StatefulWidget {
  final SharedPreferences prefs;
  const SettingsScreen({super.key, required this.prefs});
  @override
  State<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends State<SettingsScreen> {
  String _source = 'openmeteo';
  List<String> _weatherSources = ['nmc', 'openmeteo'];
  String _fontSize = 'large';
  String _refreshInterval = '30';
  bool _autoCheckUpdate = true;
  bool _useGps = true;
  String _caiyunToken = '';
  String _caiyunMode = 'none';
  String _qweatherHost = '';
  String _qweatherKey = '';

  static const _allSources = [
    ('nmc', '中国气象局', '官方国内数据，支持县区级与昨日天气'),
    ('openmeteo', 'Open-Meteo', '免费无密钥，15天预报与全球覆盖'),
    ('caiyun', '彩云天气', '分钟级降水，需凭证'),
    ('qweather', '和风天气', '需 API Host 与 Key'),
  ];

  @override
  void initState() {
    super.initState();
    _source = widget.prefs.getString('source') ?? 'openmeteo';
    final rawSources = widget.prefs.getString('weatherSources');
    if (rawSources != null && rawSources.isNotEmpty) {
      try {
        final list = (jsonDecode(rawSources) as List)
            .map((e) => e.toString())
            .where((s) => _allSources.any((a) => a.$1 == s))
            .toList();
        _weatherSources = list;
      } catch (_) {}
    } else if (widget.prefs.getString('source') != null) {
      _weatherSources = [_source];
    }
    _fontSize = widget.prefs.getString('fontSize') ?? 'large';
    _refreshInterval = widget.prefs.getString('refreshInterval') ?? '30';
    _autoCheckUpdate = widget.prefs.getBool('autoCheckUpdate') ?? true;
    _useGps = widget.prefs.getBool('useGps') ?? true;
    _caiyunToken = widget.prefs.getString('caiyunToken') ?? '';
    _caiyunMode = widget.prefs.getString('caiyunMode') ?? 'none';
    _qweatherHost = widget.prefs.getString('qweatherHost') ?? '';
    _qweatherKey = widget.prefs.getString('qweatherKey') ?? '';
  }

  Future<void> _toggleSource(String id) async {
    setState(() {
      if (_weatherSources.contains(id)) {
        _weatherSources = _weatherSources.where((s) => s != id).toList();
      } else {
        _weatherSources = [..._weatherSources, id];
      }
    });
    await widget.prefs.setString(
        'weatherSources', jsonEncode(_weatherSources));
    // 兼容旧版本：把第一优先源同步进单选键
    if (_weatherSources.isNotEmpty) {
      await widget.prefs.setString('source', _weatherSources.first);
      _source = _weatherSources.first;
    }
  }

  Future<void> _save(String key, dynamic value) async {
    if (value is String) {
      await widget.prefs.setString(key, value);
      setState(() {
        switch (key) {
          case 'source': _source = value; break;
          case 'fontSize': _fontSize = value; break;
          case 'refreshInterval': _refreshInterval = value; break;
          case 'caiyunToken': _caiyunToken = value; break;
          case 'caiyunMode': _caiyunMode = value; break;
          case 'qweatherHost': _qweatherHost = value; break;
          case 'qweatherKey': _qweatherKey = value; break;
        }
      });
    } else if (value is bool) {
      await widget.prefs.setBool(key, value);
      setState(() {
        switch (key) {
          case 'autoCheckUpdate': _autoCheckUpdate = value; break;
          case 'useGps': _useGps = value; break;
        }
      });
    }
  }

  double get _fontScale =>
      {'standard': 1.0, 'large': 1.3, 'xlarge': 1.6}[_fontSize] ?? 1.3;

  @override
  Widget build(BuildContext context) {
    // 设置页自身也随字体大小挡位缩放（对齐 native）
    return MediaQuery(
        data: MediaQuery.of(context)
            .copyWith(textScaler: TextScaler.linear(_fontScale)),
        child: Scaffold(
      appBar: AppBar(title: const Text('设置')),
      body: ListView(padding: const EdgeInsets.all(16), children: [
        _sectionTitle('天气数据源（可多选，选多个即智能聚合）'),
        ..._allSources.map((s) => _checkboxTile(
            s.$2, s.$3, _weatherSources.contains(s.$1),
            () => _toggleSource(s.$1))),
        if (_weatherSources.contains('caiyun')) ...[
          const SizedBox(height: 8),
          _sectionTitle('彩云接入方式'),
          _radioTile('V1 Token', '免费版，3天预报，Token 在 URL 中', _caiyunMode == 'v1',
              () => _save('caiyunMode', 'v1')),
          if (_caiyunMode == 'v1') ...[
            const SizedBox(height: 8),
            TextField(
                obscureText: true,
                enableSuggestions: false,
                autocorrect: false,
                decoration: const InputDecoration(
                    hintText: '填写 V1 Token', border: OutlineInputBorder()),
                controller: TextEditingController(text: _caiyunToken),
                onChanged: (v) => _save('caiyunToken', v.trim())),
            const SizedBox(height: 8),
            Text('在 dashboard.caiyunapp.com 注册获取；免费版仅 3 天预报与 48 小时逐时。',
                style: TextStyle(fontSize: 14, color: Colors.grey)),
          ],
        ],
        if (_weatherSources.contains('qweather')) ...[
          const SizedBox(height: 8),
          _sectionTitle('和风天气凭证'),
          TextField(
              enableSuggestions: false,
              autocorrect: false,
              decoration: const InputDecoration(
                  hintText: 'API Host（如 devapi.qweather.com）',
                  border: OutlineInputBorder()),
              controller: TextEditingController(text: _qweatherHost),
              onChanged: (v) => _save('qweatherHost', v.trim())),
          const SizedBox(height: 8),
          TextField(
              obscureText: true,
              enableSuggestions: false,
              autocorrect: false,
              decoration: const InputDecoration(
                  hintText: 'API Key', border: OutlineInputBorder()),
              controller: TextEditingController(text: _qweatherKey),
              onChanged: (v) => _save('qweatherKey', v.trim())),
        ],
        Text('凭证未填写完整的源会自动跳过；多选时各字段按源权重智能聚合',
            style: TextStyle(fontSize: 14, color: Colors.grey)),
        const SizedBox(height: 16),
        _sectionTitle('字体大小'),
        _radioTile('标准', '', _fontSize == 'standard',
            () => _save('fontSize', 'standard')),
        _radioTile(
            '大', '', _fontSize == 'large', () => _save('fontSize', 'large'),
            badge: '推荐'),
        _radioTile(
            '特大', '', _fontSize == 'xlarge', () => _save('fontSize', 'xlarge')),
        const SizedBox(height: 16),
        _sectionTitle('自动刷新'),
        _radioTile('关闭', '', _refreshInterval == 'off',
            () => _save('refreshInterval', 'off')),
        _radioTile('每次进入App', '', _refreshInterval == 'on_resume',
            () => _save('refreshInterval', 'on_resume')),
        _radioTile('每 10 分钟', '', _refreshInterval == '10',
            () => _save('refreshInterval', '10')),
        _radioTile('每 30 分钟', '', _refreshInterval == '30',
            () => _save('refreshInterval', '30'),
            badge: '推荐'),
        _radioTile('每 60 分钟', '', _refreshInterval == '60',
            () => _save('refreshInterval', '60')),
        _radioTile('每 6 小时', '', _refreshInterval == '360',
            () => _save('refreshInterval', '360')),
        _radioTile('每 12 小时', '', _refreshInterval == '720',
            () => _save('refreshInterval', '720')),
        _radioTile('每 24 小时', '', _refreshInterval == '1440',
            () => _save('refreshInterval', '1440')),
        const SizedBox(height: 16),
        SwitchListTile(
          title: const Text('使用当前位置'),
          subtitle: const Text('开启后会请求定位权限并自动更新天气'),
          value: _useGps,
          onChanged: (v) => _save('useGps', v),
        ),
        const SizedBox(height: 16),
        SwitchListTile(
            title: const Text('自动检查更新'),
            subtitle: const Text('进入App时检查GitHub新版本'),
            value: _autoCheckUpdate,
            onChanged: (v) => _save('autoCheckUpdate', v)),
        const SizedBox(height: 16),
        const Divider(),
        Padding(
            padding: const EdgeInsets.symmetric(vertical: 16),
            child: Column(children: [
              Text('晴暖天气 v1.2.2（实验版）',
                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
              SizedBox(height: 8),
              Text('晴暖天气：为长辈设计的简洁大字天气应用。',
                  style: TextStyle(fontSize: 16, color: Colors.grey)),
              Text('数据来源：中央气象台 / 彩云天气 / 和风天气 / Open-Meteo',
                  style: TextStyle(fontSize: 14, color: Colors.grey)),
            ])),
      ]),
        ),
    );
  }

  Widget _sectionTitle(String t) => Text(t,
      style: const TextStyle(fontSize: 22, fontWeight: FontWeight.bold));

  Widget _checkboxTile(
      String label, String subtitle, bool checked, VoidCallback onTap) {
    return CheckboxListTile(
      title: Text(label, style: const TextStyle(fontSize: 18)),
      subtitle: subtitle.isNotEmpty
          ? Text(subtitle,
              style: const TextStyle(fontSize: 14, color: Colors.grey))
          : null,
      value: checked,
      onChanged: (_) => onTap(),
      activeColor: const Color(0xFF0B6BCB),
      controlAffinity: ListTileControlAffinity.leading,
    );
  }

  Widget _radioTile(
      String label, String subtitle, bool selected, VoidCallback onTap,
      {String? badge}) {
    return RadioListTile(
      title: Row(children: [
        Text(label, style: const TextStyle(fontSize: 18)),
        if (badge != null) ...[
          SizedBox(width: 8),
          Container(
              padding: EdgeInsets.symmetric(horizontal: 8, vertical: 2),
              decoration: BoxDecoration(
                  color: Color(0xFF0B6BCB),
                  borderRadius: BorderRadius.circular(10)),
              child: Text(badge,
                  style: TextStyle(fontSize: 12, color: Colors.white))),
        ],
      ]),
      subtitle: subtitle.isNotEmpty
          ? Text(subtitle,
              style: const TextStyle(fontSize: 14, color: Colors.grey))
          : null,
      value: true,
      groupValue: selected,
      onChanged: (_) => onTap(),
      activeColor: const Color(0xFF0B6BCB),
    );
  }
}

File: app/src/main/java/com/cyanweather/app/ui/SettingsScreen.kt
package com.cyanweather.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.safeDrawing
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.windowInsetsPadding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material.icons.filled.KeyboardArrowUp
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Checkbox
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.RadioButton
import androidx.compose.material3.RadioButtonDefaults
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.foundation.layout.size
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.cyanweather.app.data.AppSettings

@Composable
fun SettingsScreen(
    settings: AppSettings,
    onBack: () -> Unit,
    onSource: (String) -> Unit,
    onWeatherSources: (List<String>) -> Unit,
    onCaiyunMode: (String) -> Unit,
    onToken: (String) -> Unit,
    onCaiyunV3Key: (String) -> Unit,
    onCaiyunV3Secret: (String) -> Unit,
    onQWeatherHost: (String) -> Unit,
    onQWeatherKey: (String) -> Unit,
    onFont: (String) -> Unit,
    onRefreshInterval: (String) -> Unit,
    onExtendedForecast: (Boolean) -> Unit,
    onExtendedDays: (Int) -> Unit,
    onGetYesterday: (Boolean) -> Unit,
    onAutoCheckUpdate: (Boolean) -> Unit,
    onUseGps: (Boolean) -> Unit,
    onOpenCityPicker: () -> Unit,
    onManualCheckUpdate: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .windowInsetsPadding(WindowInsets.safeDrawing)
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            IconButton(onClick = onBack, modifier = Modifier.width(56.dp).height(56.dp)) {
                Icon(Icons.Default.ArrowBack, contentDescription = "返回", tint = Color(0xFF333333), modifier = Modifier.width(32.dp).height(32.dp))
            }
            Text("设置", style = fst(32), fontWeight = FontWeight.Bold)
        }
        Spacer(Modifier.height(12.dp))

        // 数据源
        SettingCard {
            Text("天气数据源与优先级", style = fst(24), fontWeight = FontWeight.Bold)
            Text("勾选多个来源；越靠上优先级越高，后续来源负责补全与回退。", style = fst(16), color = Color(0xFF666666))
            Spacer(Modifier.height(6.dp))
            val labels = mapOf(
                "nmc" to ("中国气象局" to "官方国内数据，默认第一优先级"),
                "openmeteo" to ("Open-Meteo" to "免费无密钥，15天预报与全球覆盖"),
                "caiyun" to ("彩云天气" to "分钟级降水，需凭证"),
                "qweather" to ("和风天气" to "预警与丰富预报，需凭证"),
                "xiaomi" to ("小米天气" to "设备本地优先，网络接口为实验性")
            )
            val selected = settings.weatherSources.distinct()
            val display = selected + labels.keys.filterNot(selected::contains)
            display.forEach { id ->
                val info = labels.getValue(id)
                val index = selected.indexOf(id)
                SourcePriorityRow(id, info.first, info.second, index >= 0, index, selected.size,
                    onToggle = { checked ->
                        val next = if (checked) selected + id else selected - id
                        if (next.isNotEmpty()) onWeatherSources(next)
                    },
                    onMove = { delta ->
                        val target = index + delta
                        if (index >= 0 && target in selected.indices) {
                            val next = selected.toMutableList()
                            val item = next.removeAt(index)
                            next.add(target, item)
                            onWeatherSources(next)
                        }
                    })
                if (id == "qweather" && index >= 0) {
                    Spacer(Modifier.height(4.dp))
                    var qHost by remember(settings.qweatherHost) { mutableStateOf(settings.qweatherHost) }
                    var qKey by remember(settings.qweatherKey) { mutableStateOf(settings.qweatherKey) }
                    Text("和风天气凭证（紧随选项配置）", style = fst(17), fontWeight = FontWeight.Bold, color = Color(0xFF555555))
                    OutlinedTextField(value = qHost, onValueChange = { qHost = it.trim(); onQWeatherHost(qHost) },
                        placeholder = { Text("API Host，例如 abcxyz.qweatherapi.com", style = fst(16)) },
                        textStyle = MaterialTheme.typography.bodyLarge, modifier = Modifier.fillMaxWidth())
                    Spacer(Modifier.height(6.dp))
                    OutlinedTextField(value = qKey, onValueChange = { qKey = it.replace(Regex("\\s"), ""); onQWeatherKey(qKey) },
                        placeholder = { Text("API Key", style = fst(17)) }, visualTransformation = PasswordVisualTransformation(),
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                        textStyle = MaterialTheme.typography.bodyLarge, modifier = Modifier.fillMaxWidth())
                }
                if (id == "xiaomi" && index >= 0) {
                    Text("提示：优先读取小米系统天气；其他设备使用实验性网络接口，失败后自动进入下一优先级。", style = fst(15), color = Color(0xFF8A5A00))
                }
                if (id == "caiyun" && index >= 0) {
                    Spacer(Modifier.height(4.dp))
                    Text("接入方式", style = fst(17), fontWeight = FontWeight.Bold, color = Color(0xFF555555))
                    RadioRow("V1：Token 接入", "免费版，3天预报，Token 在 URL 中", settings.caiyunMode == "v1", onCaiyunMode, "v1")
                    RadioRow("V3：AppKey + AppSecret 接入", "开放平台凭证，签名鉴权", settings.caiyunMode == "v3", onCaiyunMode, "v3")
                    if (settings.caiyunMode == "v1") {
                        Spacer(Modifier.height(6.dp))
                        var token by remember { mutableStateOf(settings.caiyunToken) }
                        OutlinedTextField(
                            value = token,
                            onValueChange = { token = it.replace(Regex("\\s"), ""); onToken(token) },
                            placeholder = { Text("填写 V1 Token", style = fst(18)) },
                            textStyle = MaterialTheme.typography.bodyLarge,
                            visualTransformation = PasswordVisualTransformation(),
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                            modifier = Modifier.fillMaxWidth()
                        )
                        Spacer(Modifier.height(8.dp))
                        Text("在 dashboard.caiyunapp.com 注册获取；免费版仅 3 天预报与 48 小时逐时。", style = fst(16), color = Color(0xFF666666))
                    }
                    if (settings.caiyunMode == "v3") {
                        Spacer(Modifier.height(6.dp))
                        var key by remember { mutableStateOf(settings.caiyunV3Key) }
                        var secret by remember { mutableStateOf(settings.caiyunV3Secret) }
                        OutlinedTextField(
                            value = key,
                            onValueChange = { key = it.replace(Regex("\\s"), ""); onCaiyunV3Key(key) },
                            placeholder = { Text("填写 AppKey", style = fst(18)) },
                            textStyle = MaterialTheme.typography.bodyLarge,
                            visualTransformation = PasswordVisualTransformation(),
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                            modifier = Modifier.fillMaxWidth()
                        )
                        Spacer(Modifier.height(8.dp))
                        OutlinedTextField(
                            value = secret,
                            onValueChange = { secret = it.replace(Regex("\\s"), ""); onCaiyunV3Secret(secret) },
                            placeholder = { Text("填写 AppSecret", style = fst(18)) },
                            textStyle = MaterialTheme.typography.bodyLarge,
                            visualTransformation = PasswordVisualTransformation(),
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                            modifier = Modifier.fillMaxWidth()
                        )
                        Spacer(Modifier.height(8.dp))
                        Text("在彩云开放平台获取；使用 AppKey 与 AppSecret 生成签名访问接口。", style = fst(16), color = Color(0xFF666666))
                    }
                    Text("凭证未填写完整时，自动使用中国气象局数据。", style = fst(16), color = Color(0xFF666666))
                }
            }
        }

        // 扩展多日预报（仅彩云天气凭证已填写时）
        if ("caiyun" in settings.weatherSources && settings.caiyunMode == "v1" && settings.caiyunToken.isNotBlank() ||
            "caiyun" in settings.weatherSources && settings.caiyunMode == "v3" && settings.caiyunV3Key.isNotBlank()) {
            SettingCard {
                SwitchRow(
                    "扩展多日预报",
                    "彩云天气免费版单次最多请求3天数据，多次请求叠加更多日期",
                    settings.extendedForecast,
                    onExtendedForecast
                )
                if (settings.extendedForecast) {
                    Spacer(Modifier.height(6.dp))
                    SwitchRow("获取昨日天气", "开启后从昨天开始获取，昨日计入总天数", settings.getYesterday, onGetYesterday)
                    Spacer(Modifier.height(6.dp))
                    val days = settings.extendedDays
                    val totalDays = days + if (settings.getYesterday) 1 else 0
                    val requests = (days + 2) / 3 + if (settings.getYesterday) 1 else 0
                    Text("总获取天数：${totalDays}天，共需 ${requests} 次请求", style = fst(18), color = Color(0xFF666666))
                    Spacer(Modifier.height(6.dp))
                    RadioRow("3天", "", days == 3, { it.toIntOrNull()?.let(onExtendedDays) }, "3")
                    RadioRow("5天", "", days == 5, { it.toIntOrNull()?.let(onExtendedDays) }, "5")
                    RadioRow("7天", "", days == 7, { it.toIntOrNull()?.let(onExtendedDays) }, "7")
                    RadioRow("10天", "", days == 10, { it.toIntOrNull()?.let(onExtendedDays) }, "10")
                    RadioRow("15天", "", days == 15, { it.toIntOrNull()?.let(onExtendedDays) }, "15")
                }
            }
        }

        // 城市
        SettingCard {
            Text("城市", style = fst(24), fontWeight = FontWeight.Bold)
            Spacer(Modifier.height(8.dp))
            Card(
                onClick = onOpenCityPicker,
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
            ) {
                Row(
                    Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(settings.cityName, style = fst(24), modifier = Modifier.weight(1f))
                    Text("更改 >", style = fst(24), color = Color(0xFF0B6BCB))
                }
            }
            Spacer(Modifier.height(6.dp))
            Text(
                if (settings.source in listOf("openmeteo", "caiyun", "qweather", "smart")) "该数据源使用定位获取，无需选择城市。"
                else "支持省份 → 城市 / 区县 两级选择。",
                style = fst(16),
                color = Color(0xFF666666)
            )
        }

        // 字号
        SettingCard {
            Text("字体大小", style = fst(24), fontWeight = FontWeight.Bold)
            Spacer(Modifier.height(6.dp))
            RadioRow("标准", "", settings.fontSize == "standard", onFont, "standard")
            RadioRow("大", "", settings.fontSize == "large", onFont, "large", badge = "推荐")
            RadioRow("特大", "", settings.fontSize == "xlarge", onFont, "xlarge")
        }

        // 自动定位
        SettingCard {
            SwitchRow("使用自动定位", "智能组合、彩云、和风与 Open-Meteo 按定位获取；需要定位权限", settings.useGps, onUseGps)
        }

        // 自动刷新
        SettingCard {
            Text("自动刷新", style = fst(24), fontWeight = FontWeight.Bold)
            Spacer(Modifier.height(6.dp))
            RadioRow("关闭", "", settings.refreshInterval == "off", onRefreshInterval, "off")
            RadioRow("每次进入App", "", settings.refreshInterval == "on_resume", onRefreshInterval, "on_resume")
            RadioRow("每 10 分钟", "", settings.refreshInterval == "10", onRefreshInterval, "10")
            RadioRow("每 30 分钟", "", settings.refreshInterval == "30", onRefreshInterval, "30", badge = "推荐")
            RadioRow("每 60 分钟", "", settings.refreshInterval == "60", onRefreshInterval, "60")
            RadioRow("每 6 小时", "", settings.refreshInterval == "360", onRefreshInterval, "360")
            RadioRow("每 12 小时", "", settings.refreshInterval == "720", onRefreshInterval, "720")
            RadioRow("每 24 小时", "", settings.refreshInterval == "1440", onRefreshInterval, "1440")
        }

        // 检查更新
        SettingCard {
            SwitchRow("自动检查更新", "进入App时检查GitHub是否有新版本", settings.autoCheckUpdate, onAutoCheckUpdate)
            Spacer(Modifier.height(8.dp))
            Card(
                onClick = onManualCheckUpdate,
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
            ) {
                Row(
                    Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text("手动检查更新", style = fst(24), modifier = Modifier.weight(1f))
                    Text("检查 >", style = fst(24), color = Color(0xFF0B6BCB))
                }
            }
        }

        // 关于
        SettingCard {
            val context = LocalContext.current
            val versionName = remember {
                runCatching {
                    context.packageManager.getPackageInfo(context.packageName, 0).versionName
                }.getOrNull() ?: ""
            }
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text("关于", style = fst(24), fontWeight = FontWeight.Bold)
                if (versionName.isNotEmpty()) {
                    Spacer(Modifier.width(8.dp))
                    Text("v$versionName", style = fst(16), color = Color(0xFF888888))
                }
            }
            Spacer(Modifier.height(8.dp))
            Text("晴暖天气：为长辈设计的简洁大字天气应用。", style = fst(18), color = Color(0xFF666666))
            Text("数据来源：中央气象台 / 彩云天气 / 和风天气 / Open-Meteo / 小米天气", style = fst(18), color = Color(0xFF666666))
        }
        Spacer(Modifier.height(24.dp))
    }
}

@Composable
private fun SettingCard(content: @Composable () -> Unit) {
    Card(
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        modifier = Modifier.fillMaxWidth().padding(bottom = 14.dp)
    ) {
        Column(Modifier.padding(16.dp)) { content() }
    }
}

@Composable
private fun SourcePriorityRow(
    id: String,
    label: String,
    hint: String,
    checked: Boolean,
    index: Int,
    selectedCount: Int,
    onToggle: (Boolean) -> Unit,
    onMove: (Int) -> Unit
) {
    Row(
        Modifier.fillMaxWidth().padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Checkbox(checked = checked, onCheckedChange = onToggle)
        Column(Modifier.weight(1f)) {
            Text(label, style = fst(21))
            Text(hint, style = fst(15), color = Color(0xFF666666))
        }
        if (checked) {
            Text("优先级 ${index + 1}", style = fst(14), color = Color(0xFF0B6BCB))
            IconButton(
                onClick = { onMove(-1) }, enabled = index > 0,
                modifier = Modifier.size(38.dp)
            ) {
                Icon(Icons.Default.KeyboardArrowUp, contentDescription = "上移", tint = if (index > 0) Color(0xFF0B6BCB) else Color(0xFFBDBDBD))
            }
            IconButton(
                onClick = { onMove(1) }, enabled = index < selectedCount - 1,
                modifier = Modifier.size(38.dp)
            ) {
                Icon(Icons.Default.KeyboardArrowDown, contentDescription = "下移", tint = if (index < selectedCount - 1) Color(0xFF0B6BCB) else Color(0xFFBDBDBD))
            }
        }
    }
}

@Composable
private fun RadioRow(
    label: String,
    hint: String,
    selected: Boolean,
    onSelect: (String) -> Unit,
    value: String,
    badge: String? = null
) {
    Row(
        Modifier.fillMaxWidth().padding(vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        RadioButton(
            selected = selected,
            onClick = { onSelect(value) },
            colors = RadioButtonDefaults.colors(selectedColor = Color(0xFF0B6BCB))
        )
        Column {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(label, style = fst(22))
                if (badge != null) {
                    Spacer(Modifier.width(8.dp))
                    Box(
                        Modifier
                            .clip(RoundedCornerShape(50))
                            .background(Color(0xFF0B6BCB))
                            .padding(horizontal = 10.dp, vertical = 2.dp)
                    ) {
                        Text(badge, style = fst(14), color = Color.White)
                    }
                }
            }
            if (hint.isNotEmpty()) {
                Text(hint, style = fst(16), color = Color(0xFF666666))
            }
        }
    }
}

@Composable
private fun SwitchRow(
    label: String,
    hint: String,
    checked: Boolean,
    onChecked: (Boolean) -> Unit
) {
    Row(
        Modifier.fillMaxWidth().padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column(Modifier.weight(1f)) {
            Text(label, style = fst(22))
            if (hint.isNotEmpty()) {
                Text(hint, style = fst(16), color = Color(0xFF666666))
            }
        }
        Switch(checked = checked, onCheckedChange = onChecked)
    }
}
